android/src/main/java/com/mapbox/reactnativemapboxgl/RNMGLAnnotationOptionsFactory.java

```diff
@@ -0,0 +1,225 @@
+        int resID = context.getResources().getIdentifier(drawableName, "drawable", context.getApplicationContext().getPackageName());
+        return ContextCompat.getDrawable(context, resID);
+    }
+
+    static Drawable drawableFromUrl(Context context, String url) throws IOException {
+        // This doesn't currently work, as it throws NetworkOnMainThreadException
+        HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
+        connection.connect();
+        InputStream input = connection.getInputStream();
+
+        Bitmap bitmap = BitmapFactory.decodeStream(input);
+        return new BitmapDrawable(context.getResources(), bitmap);
+    }
+
+    static Map<String, Icon> iconCache = new HashMap();
+
+    static Icon iconFromPathAndSize(Context context, String path, int width, int height) throws IOException {
+        String cacheKey = path + "||" + width + "||" + height;
+        Icon icon = iconCache.get(cacheKey);
+        if (icon != null) { return icon; }
+
+        Drawable drawable;
+        if (path.startsWith("image!")) {
+            drawable = drawableFromDrawableName(context, path.replace("image!", ""));
+        } else {
+            drawable = drawableFromUrl(context, path);
+        }
+
+        IconFactory iconFactory = IconFactory.getInstance(context);
+
+        int intrinsicWidth = drawable.getIntrinsicWidth();
+        int intrinsicHeight = drawable.getIntrinsicHeight();
+
+        if (width < 0) { width = intrinsicWidth; }
+        if (height < 0) { height = intrinsicHeight; }
+
+        // Check if a rescale would be superfluous
+        if ((drawable instanceof BitmapDrawable) && width == intrinsicWidth && height == intrinsicHeight) {
+            icon = iconFactory.fromBitmap(((BitmapDrawable)drawable).getBitmap());
+        } else {
+            icon = iconFactory.fromDrawable(drawable, width, height);
+        }
+
+        iconCache.put(cacheKey, icon);
+        return icon;
+    }
+
+    static RNMGLAnnotationOptions markerOptionsFromJS(ReadableMap annotation, Context context) {
+        MarkerOptions marker = new MarkerOptions();
+
+        double latitude = annotation.getArray("coordinates").getDouble(0);
+        double longitude = annotation.getArray("coordinates").getDouble(1);
+        LatLng markerCenter = new LatLng(latitude, longitude);
+        marker.position(markerCenter);
+
+        if (annotation.hasKey("title")) {
+            String title = annotation.getString("title");
+            marker.title(title);
+        }
+
+        if (annotation.hasKey("subtitle")) {
+            String subtitle = annotation.getString("subtitle");
+            marker.snippet(subtitle);
+        }
+
+        if (annotation.hasKey("annotationImage")) {
+            ReadableMap annotationImage = annotation.getMap("annotationImage");
+            String annotationURL = annotationImage.getString("url");
+            try {
+                int width = -1;
+                int height = -1;
```

Comment:
Just a small convention to say "intrinsic size of the drawable". See L117 above.